File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/calculo/ferias/regras/AbonoDeFeriasValidRule.java
/*
 * Decompiled with CFR 0.152.
 */
package br.jus.trt8.pjecalc.negocio.dominio.calculo.ferias.regras;

import br.jus.trt8.pjecalc.base.comum.Utils;
import br.jus.trt8.pjecalc.base.comum.validadores.ValidatorContext;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidRule;
import br.jus.trt8.pjecalc.negocio.comum.validators.ValidValueValidator;
import br.jus.trt8.pjecalc.negocio.constantes.Mensagens;
import br.jus.trt8.pjecalc.negocio.constantes.SituacaoDaFeriasEnum;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.ferias.Ferias;

public class AbonoDeFeriasValidRule
implements ValidRule {
    @Override
    public Mensagens getMessage() {
        return Mensagens.MSG0004;
    }

    @Override
    public boolean isValid(ValidValueValidator validator, ValidatorContext context) {
        Boolean abono = (Boolean)context.getValue();
        if (Utils.nulo(abono)) {
            return true;
        }
        Ferias ferias = (Ferias)context.getBean();
        if (abono.booleanValue()) {
            return SituacaoDaFeriasEnum.GOZADAS.equals((Object)ferias.getSituacao()) || SituacaoDaFeriasEnum.GOZADAS_PARCIALMENTE.equals((Object)ferias.getSituacao());
        }
        return true;
    }
}

File: src/lib/pjecalc/core/dominio/calculo/ferias/regras/abono-de-ferias-valid-rule.ts
/**
 * Porte 1:1 de AbonoDeFeriasValidRule.java (34 linhas).
 *
 * Regra: abono pecuniário só pode ser marcado se a situação for GOZADAS ou
 * GOZADAS_PARCIALMENTE (férias indenizadas ou perdidas não têm abono).
 *
 * Ref: pjecalc-fonte/.../dominio/calculo/ferias/regras/AbonoDeFeriasValidRule.java
 */
import { SituacaoDaFeriasEnum } from '../../../../constantes/enums';
import type { Ferias } from '../ferias';

export class AbonoDeFeriasValidRule {
  static readonly MESSAGE = 'MSG0004';

  static isValid(abono: boolean | null, ferias: Ferias): boolean {
    if (abono === null) return true;
    if (abono) {
      return ferias.getSituacao() === SituacaoDaFeriasEnum.GOZADAS
        || ferias.getSituacao() === SituacaoDaFeriasEnum.GOZADAS_PARCIALMENTE;
    }
    return true;
  }
}
